modules/utils/src/merkle.ts

```diff
@@ -1,26 +1,36 @@
+import * as merkle from "vector-merkle-tree";
 import { CoreTransferState } from "@connext/vector-types";
 import { HashZero } from "@ethersproject/constants";
 import { keccak256 } from "ethereumjs-util";
 import { MerkleTree } from "merkletreejs";
 
-import { hashCoreTransferState } from "./transfers";
-
-export const generateMerkleTreeData = (transfers: CoreTransferState[]): { root: string; tree: MerkleTree } => {
-  // Sort transfers alphabetically by id
-  const sorted = transfers.sort((a, b) => a.transferId.localeCompare(b.transferId));
+import { encodeCoreTransferState, hashCoreTransferState } from "./transfers";
 
+export const generateMerkleTreeData = (transfers: CoreTransferState[]): { root: string; tree: merkle.Tree } => {
   // Create leaves
-  const leaves = sorted.map((transfer) => {
-    return hashCoreTransferState(transfer);
+  const tree = new merkle.Tree();
+  tree.free(); // handle memory leaks
+  transfers.forEach((transfer) => {
+    tree.insert_hex_js(encodeCoreTransferState(transfer));
   });
 
-  // Generate tree
-  const tree = new MerkleTree(leaves, keccak256, { sortPairs: true });
-
   // Return
-  const calculated = tree.getHexRoot();
+  const calculated = tree.root_js();
+
```

Comment:
Do not return the `tree` object once it has been `free`'d. Any method call will throw an exception.